### DISCUSSION TITLE
Many properties within SteamPending are readonly, despite Provider.pending being public static.

### DISCUSSION THREAD

**Discussion by Jdance-Media** [SmartlyDressedGames/Unturned-3.x-Community#4174] [Ideas]
There are several plugins out there that modify a character's appearance when they first join the server. There are currently two ways of going about this:

1. Using reflection to modify the private fields.
2. Making an entire new SteamPending and copying over the data using its constructor. Afterwards, delete the old entry from Provider.pending and add the new one.

If the player can make a new SteamPending to modify properties like color, face, beard, and etc. then it seems pointless for those properties to be readonly, as it only creates a hassle for plugin developers and doesn't really totally prevent modifying them.

**Comment by SDGNelson** [comment]
It never should have been public in the first place. Unfortunately, when I was younger I defaulted to making everything public without a good reason. The code I've written in the last few years has better distinction between what should be public for plugins vs default private or internal.

**Comment by Jdance-Media** [comment]
Ah okay. Makes sense. Will avoid using Provider.pending and SteamPending then considering that it might be subject to change.
